cmd/haproxy-cfg-generator/cfg_generator.go

```diff
@@ -196,13 +197,16 @@ func get_backends(config *Config) string {
 
 	result := ""
 
-	rp_backend := "backend tenant_api_%v\n    server tp_%v %v:%v maxconn %v\n\n"
+	//TODO: add flag to the tool for secure port or insecure port usage
+	//
+	tp_backend := "backend tenant_api_%v\n    server tp_%v %v:%v maxconn %v ssl check ca-file /etc/haproxy/ca.crt\n\n"
+
 	for index, ip := range config.tenantPartititonIPs {
-		result = result + fmt.Sprintf(rp_backend, index+1, index+1, ip, ARKTOS_API_PORT, MAX_CONN_PER_BACKEND)
+		result = result + fmt.Sprintf(tp_backend, index+1, index+1, ip, ARKTOS_API_SECURE_PORT, MAX_CONN_PER_BACKEND)
```

Comment:
Please make sure this works with scale up local cluster setup before merge to the poc branch

Reply:
this has been tested with both scale up and scale out with Kubemark. this should be a black box to the kubemark or onebox setup anyways.